internal/services/register.go

```diff
@@ -1,18 +1,155 @@
 package services
 
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/shulganew/gophermart/internal/config"
+	"github.com/shulganew/gophermart/internal/model"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/bcrypt"
+)
+
 // User registration service
 type Register struct {
+	stor Registrar
 }
 
 type Registrar interface {
-	Regisger() error
+	Register(ctx context.Context, user model.User) error
+	GetByLogin(ctx context.Context, login string) (*model.User, error)
+}
+
+func NewRegister(stor Registrar) *Register {
+	return &Register{stor: stor}
+}
+
+// Register new user in market
+func (r *Register) NewUser(ctx context.Context, user model.User) (userID *uuid.UUID, existed bool, err error) {
```

Comment:
```suggestion
func (r *Register) AddUser(ctx context.Context, user model.User) (userID *uuid.UUID, err error) {
```
New используется чтобы показать что метод является конструктором
Существование пользователя не нужно возвращать отдельно - можно использовать ошибку для этого

Reply:
Сделал CreateUser